server/sonar-server/src/main/java/org/sonar/server/computation/measure/MeasureVariations.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.computation.measure;
+
+import com.google.common.base.Objects;
+import java.util.Arrays;
+import javax.annotation.concurrent.Immutable;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Predicates.notNull;
+import static com.google.common.collect.FluentIterable.from;
+
+@Immutable
+public final class MeasureVariations {
+  private final Double[] variations = new Double[5];
+
+  public MeasureVariations(Double... variations) {
+    checkArgument(variations.length <= 5, "There can not be more than 5 variations");
+    checkArgument(!from(Arrays.asList(variations)).filter(notNull()).isEmpty(), "There must be at least one variation");
+    System.arraycopy(variations, 0, this.variations, 0, variations.length);
+  }
+
+  public boolean hasVariation1() {
```

Comment:
since there is only 5 variations, I think the use of a index is poor design as nothing declares the limitations of the number of variations
granted, it can make some pieces of code shorter to have the index, but I don't think it's worth it
I'm not even found of the constructor with a vararg but providing 5 overrides of it was way uglier